test/test_ion_text.cpp

```diff
@@ -720,3 +720,426 @@ TEST(IonTextInt, BinaryLiterals) {
+    ASSERT_EQ(offset, 10);
+    ASSERT_EQ(line, 3);
+    ASSERT_EQ(col_offset, 3);
+
+    // field c
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 18);
+    ASSERT_EQ(line, 5);
+    ASSERT_EQ(col_offset, 0);
+
+    // finally, step out
+    ION_ASSERT_OK(ion_reader_step_out(reader));
+
+    // read one value to ensure position is still correct after stepping out
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 22);
+    ASSERT_EQ(line, 6);
+    ASSERT_EQ(col_offset, 2);
+}
+
+/** Tests ion_reader_get_value_position for top-level values appearing on separate lines. */
+TEST(IonTextPosition, MultilinePositions) {
+    const char *ion_text = "1\n2\n3\n\n4\n";
+
+    hREADER  reader;
+    ION_TYPE type;
+
+    ION_ASSERT_OK(ion_test_new_text_reader(ion_text, &reader));
+
+    int64_t offset = 0;
+    int32_t line = 0;
+    int32_t col_offset = 0;
+    int32_t int32;
+
+    // Advance to "1", read it and assert value and position.
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ION_ASSERT_OK(ion_reader_read_int32(reader, &int32));
+    ASSERT_EQ(1, int32);
+
+    // Read the position of the int and verify we have the correct position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 0);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 0);
+
+    // Advance to "2", read it and assert value and position.
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ION_ASSERT_OK(ion_reader_read_int32(reader, &int32));
+    ASSERT_EQ(2, int32);
+    // Read the position of the int and verify we have the correct position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 2);
+    ASSERT_EQ(line, 2);
+    ASSERT_EQ(col_offset, 0);
+
+    // Advance to "3", read it and assert value and position.
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ION_ASSERT_OK(ion_reader_read_int32(reader, &int32));
+    ASSERT_EQ(3, int32);
+    // Read the position of the int and verify we have the correct position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 4);
+    ASSERT_EQ(line, 3);
+    ASSERT_EQ(col_offset, 0);
+
+    // Advance to "4", read it and assert value and position.
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ION_ASSERT_OK(ion_reader_read_int32(reader, &int32));
+    ASSERT_EQ(4, int32);
+    // Read the position of the int and verify we have the correct position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 7);
+    ASSERT_EQ(line, 5);
+    ASSERT_EQ(col_offset, 0);
+}
```

Comment:
Another interesting test case may be to make sure the offsets work properly with long strings, which may have newlines and/or comments between fragments. E.g.

```
'''foo'''
// bar
'''baz'''
```
which is equivalent to `"foobaz"`.

Reply:
I was certain there'd be a bug here, but in fact, there wasn't.